keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,298 @@
+        hidden_dim: Int. The size of the pooler layer.
+        name: String, optional. Name of the model.
+        trainable: Boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```
+    python
+    # Randomly initialized Roberta encoder
+    encoder = keras_nlp.models.Roberta(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call classifier on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=50265),
+        "input_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    classifier = keras_nlp.models.RobertaClassifier(encoder, 4)
+    logits = classifier(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        base_model,
+        num_classes,
+        hidden_dim=None,
+        dropout=0.1,
+        name=None,
+        trainable=True,
+    ):
+        inputs = base_model.input
+        if hidden_dim is None:
+            hidden_dim = base_model.hidden_dim
+
+        x = base_model(inputs)["sequence_output"][
+            :, base_model.start_token_index, :
+        ]
+        x = keras.layers.Dropout(dropout, name="pooled_dropout")(x)
+        x = keras.layers.Dense(
+            hidden_dim, activation="tanh", name="pooled_dense"
+        )(x)
+        x = keras.layers.Dropout(dropout, name="classifier_dropout")(x)
+        outputs = keras.layers.Dense(
+            num_classes,
+            kernel_initializer=_roberta_kernel_initializer(),
+            name="logits",
+        )(x)
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs=inputs, outputs=outputs, name=name, trainable=trainable
+        )
+        # All references to `self` below this line
+        self.base_model = base_model
+        self.num_classes = num_classes
+
+
+def RobertaBase(name=None, trainable=True):
+    """RoBERTa implementation using "Base" architecture.
+
+    This network implements a bi-directional Transformer-based encoder as
+    described in ["RoBERTa: A Robustly Optimized BERT Pretraining
+    Approach"](https://arxiv.org/abs/1907.11692). It includes the
+    embedding lookups and transformer layers, but not the masked language model
+    or classification task networks.
+
+    Args:
+        name: String, optional. Name of the model.
+        trainable: Boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```python
+    # Randomly initialized RobertaBase encoder
+    encoder = keras_nlp.models.RobertaBase()
+
+    # Call encoder on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 512), dtype=tf.int64, maxval=encoder.vocabulary_size),
+        "input_mask": tf.constant(
```

Comment:
tf.ones((1, 512)) would be slightly cleaner

Reply:
I was trying to create an actual vocab here to give users some idea, although now I see that it doesn't contain the right protected tokens (in this case `<s> A <\s> <\s> B <\s>`). Should we drop the pretense everywhere?